api/core.py

```diff
@@ -416,7 +430,11 @@ def _archivestream(self, ticket):
         stream = cStringIO.StringIO()
         with tarfile.open(mode='w|', fileobj=stream) as archive:
             for filepath, arcpath, _ in ticket['target']:
-                yield archive.gettarinfo(filepath, arcpath).tobuf()
+                try:
+                    yield archive.gettarinfo(filepath, arcpath).tobuf()
+                except Exception as e:
```

Comment:
Are there any scenarios other than scitran database or filesystem corruption (or a sysadmin deleting files) where this error is likely to occur? Maybe I'm missing something?

I feel super confused right now, because I'd never imagine that when downloading something from a data host I'd have to extract `did-it-work-lol.txt` and check to see if I randomly didn't get the files I asked for. I want to make sure I'm not fundamentally misunderstanding what this change does.